### DISCUSSION TITLE
[Problem]: When added to agent funtion its not calling it, intiated  Suggested tool Call after pressing enter then executing it

### DISCUSSION THREAD

**Discussion by beckortikov** [microsoft/autogen#2253] [Q&A]
Added function to agent, but it can execute it via proxy, its just typinh Suggested tool Callm after pressing enter working, how can resolve it? 
In user_proxy human_input_mode still ALWAYS, 
***** Suggested tool Call (call_rhSvZNKN1acCiomqnGnjX4ZI): available_dates *****
Arguments:
{}
********************************************************************************

--------------------------------------------------------------------------------
Provide feedback to helper. Press enter to skip and use auto-reply, or type 'exit' to end the conversation:

**Comment by ekzhu** [comment]
This issue https://github.com/microsoft/autogen/issues/2223 propose an agent that can self-executing tools.

For now tool calls and code blocks are executed in a separate agent. You can read our tutorial: https://microsoft.github.io/autogen/docs/tutorial/tool-use
